python/llm/src/bigdl/llm/transformers/utils.py

```diff
@@ -224,37 +224,49 @@ def load_imatrix_data(imatrix_file):
     return imatrix_data
 
 
+def module_name_process(full_module_name):
+    # full name maybe model.layers.31.self_attn.o_proj
+    # TODO: how to better aligned and generalize
+    module_name = full_module_name.split('.')
+    if len(module_name) == 5:
+        layer = module_name[2]
+        cur_module = module_name[-1][:-5]
+        new_module_name = '_'.join([layer, cur_module])
+    elif len(module_name) == 1:
+        new_module_name = module_name[0]
+        layer = None
+        cur_module = None
+    return new_module_name, layer, cur_module
+
+
 def get_cur_qtype_and_imatrix(qtype, full_module_name, imatrix_data):
-    if qtype in [ggml_tensor_qtype["iq2_xxs"], ggml_tensor_qtype["iq2_xs"],
-                 ggml_tensor_qtype["q2_k"]] and imatrix_data is not None:
+    cur_qtype = qtype
+    if qtype in [ggml_tensor_qtype["iq2_xxs"], ggml_tensor_qtype["iq2_xs"]]:
         # For quantization which needs importance matrix
-        # module name preprocess
-        # full name maybe model.layers.31.self_attn.o_proj
-        # TODO: just consider llama/mistral here
-        # TODO: how to better aligned and generalize
-        module_name = full_module_name.split('.')
-        cur_qtype = qtype
-        if len(module_name) == 5:
-            layer = module_name[2]
-            cur_module = module_name[-1][:-5]
-            new_module_name = '_'.join([layer, cur_module])
-        elif len(module_name) == 1:
-            new_module_name = module_name[0]
-            layer = None
-            cur_module = None
+        new_module_name, layer, cur_module = module_name_process(full_module_name)
+        # custom mixed quantization strategy
+        if cur_module == 'v' or (cur_module == 'down' and int(layer) in [0, 1, 10, 11]):
+            cur_qtype = ggml_tensor_qtype['q2_k']
         if imatrix_data is not None and new_module_name in imatrix_data:
             cur_imatrix = imatrix_data[new_module_name]
-            # custom mixed quantization strategy
-            if cur_module == 'v' or (cur_module == 'down' and int(layer) in [0, 1, 10, 11]) \
-                    or new_module_name == 'lm_head':
-                cur_qtype = ggml_tensor_qtype['sym_int4']
         else:
+            # if no imatrix is available, use fp8 for lm_head
             cur_imatrix = None
-            # custom mixed quantization strategy
-            if cur_module == 'v' or (cur_module == 'down' and int(layer) in [0, 1, 10, 11]) \
-                    or new_module_name == 'lm_head':
-                cur_qtype = ggml_tensor_qtype['sym_int4']
+            if new_module_name == 'lm_head':
+                cur_qtype = ggml_tensor_qtype['fp8']
```

Comment:
shall we use int8?

Reply:
yeah we can use int8, I have verified locally that int8 can produce similar output as fp8.